content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+We have three common generic collections incorporated in the C# namespace.
+
+## 1. LIST
+
+A list is the most commonly used generic collection. Its because they are much more convenient when they are used instead of an array.
+A list is a very general data structure with each entry occupying a specific position known as an index. Lists are very flexible as they automatically resize every time data is added.
+
+Lists are very easy to manipulate. This is because many methods have been created and abstracted, meaning the bulky work is already done for the user. The user only has to call the method and manipulate the list.
+ 
+A list stores data in sequence, although the sequence oughtn't to have to be particular. It can also accept duplicate entries.
+Below is a pictorial explanation of a list. A generic list class is declared as shown:
+
+```
+public class genericList<T>
+
+{
+    
+    void add(T input){}
+}
+```
+
+On the same note, a list can be of integer or string data type. They are declared differently but the syntax is the same as shown:
+```
+using system;
+
+{
+private class genericListExample
+static void Main ()
+{
+    genericlist<int> list1 = new genericlist<int>(); //This declares a  list of type int.
+    
+     genericlist<string> list2 = new genericlist<string>(); //This declares a list of type string.
+     
+      genericlist<int> list1 = new genericlist<int>(); //This declares a list of type int.
+}
+}
+```
+
+A simple c# example of generic list implementation might help in understanding.
+
+```
+using system;
+using system.collection.generic; // A system namespace for implementing generic lists.
+
+namespace GenericListExample
+{
+    public class Generic<T>  // Declare Generics.
+    {
+        public void GenFunction(T printvalue)
+        {
+            Console.WriteLine(printvalue)
+        }
+    }
+    public class program
+    {
+        public static void Main(string [] args)
+        {
+            Console.WriteLine("Printing integer value");
+            Generic<int> gen = new Generic<int>();
+            gen.GenFunction(10);
+            
+            Console.WriteLine("Printing string value");
+            Generic<string> genString = new Generic<string>();
+            gen.GenFunction("Hello world");
+            
+        }
+    }
+}
+```
+
+Note that _<T>_ is a placeholder for data types and replaced by datatype at compile time.
+
+### 2. Hashsets
+This is an implementation of a set. A set is like a big bag of entries with no particular order rather than a list. They are based on the discrete mathematic's idea of a 'set', which is the grouping of entries that can contain one, two, or many distinct items.  Sorting in a hash set is impossible because the entries have to be sorted. Use of a hash set is very much applicable where we have to store a large amount of data that doesn't have to be sorted eg storing the identity of cars passing through an entrance gate. A program example:
+
+```
```

Comment:
> ```
>  using System;
> using System.Collections.Generic;
>   
> class MyHashSet {
>     static public void Main()
>     {
>   
>         // Creating HashSet
>      
>         HashSet<string> myhash1 = new HashSet<string>();
>   
>         // Add the elements in HashSet
>   
>         myhash1.Add("Ndung'u");
>         myhash1.Add("Elijah");
>         // Creating another HashSet
>      
>         HashSet<string> myhash2 = new HashSet<string>();
>         myhash2.Add("Section");
>         myhash2.Add("Git");
>    
>   
>         // Using ExceptWith method
>         myhash1.ExceptWith(myhash2);
>         foreach(var ele in myhash1)
>         {
>             Console.WriteLine(ele);
>         }
>     }
> }
> ```

This is missing syntax highlighting, see the comment on your first code example (the genericList class) for details.